packages/apps/dashboard/server/src/modules/details/details.service.ts

```diff
@@ -147,68 +148,40 @@ export class DetailsService {
     return result;
   }
 
-  public async getBestLeadersByRole(chainId?: ChainId): Promise<LeaderDto[]> {
+  public async getLeadersByChainId(
+    chainId?: ChainId,
+    take?: number,
+  ): Promise<LeaderDto[]> {
     const chainIds = !chainId
       ? await this.networkConfig.getAvailableNetworks()
       : [chainId];
 
-    const leadersByRole: { [role: string]: LeaderDto } = {};
-
+    let allLeadersData: ILeader[] = [];
     for (const id of chainIds) {
       const leadersData = await OperatorUtils.getLeaders({ chainId: id });
-
-      for (const leaderData of leadersData) {
-        const leaderDto: LeaderDto = plainToInstance(LeaderDto, leaderData, {
-          excludeExtraneousValues: true,
-        });
-        leaderDto.chainId = id;
-
-        const role = leaderDto.role;
-
-        if (Object.values(Role).includes(role)) {
-          if (
-            !leadersByRole[role] ||
-            BigInt(leaderDto.amountStaked) >
-              BigInt(leadersByRole[role].amountStaked)
-          ) {
-            leadersByRole[role] = leaderDto;
-          }
-        }
-      }
+      allLeadersData = allLeadersData.concat(
+        leadersData.filter((leader) => leader.amountStaked > 0 && leader.role),
+      );
     }
 
-    const reputations = await this.fetchReputations();
-    this.assignReputationsToLeaders(Object.values(leadersByRole), reputations);
-
-    return Object.values(leadersByRole);
-  }
-
-  public async getAllLeaders(chainId?: ChainId): Promise<LeaderDto[]> {
-    const chainIds = !chainId
-      ? await this.networkConfig.getAvailableNetworks()
-      : [chainId];
-
-    const allLeaders: LeaderDto[] = [];
+    allLeadersData = allLeadersData.sort((a, b) =>
```

Comment:
You don't have to re-assign value when calling `.sort`, it does sorting in place
```suggestion
    allLeadersData.sort((a, b) =>
```